docs-website/sphinx/conf.py

```diff
@@ -26,8 +26,10 @@
     "sphinx_autodoc_typehints",
     # This enables us to autogenerate docs for our CLI.
     "sphinx_click",
+    "sphinx_markdown_builder",
 ]
 
+markdown_anchor_sections = True
```

Comment:
<img width="579" alt="image" src="https://github.com/user-attachments/assets/48cb1aae-4bc1-446d-bf55-f36fb4bdee87" />

1. the blockiness of the str / Optional etc words feels extremely ugly? maybe we should get rid of the border to make this look more text-like?
2. the trailing `-` still looks really ugly - can we get rid of that?